nano/node/repcrawler.cpp

```diff
@@ -324,6 +324,10 @@ void nano::rep_crawler::cleanup_reps ()
 				equal = true;
 			}
 		}
+		else if (i->get_type () == nano::transport::transport_type::fake)
+		{
+			equal = true;
+		}
```

Comment:
@dsiganos I believe repcrawler is currently broken for inproc channels too. We should set equal to true for inproc channels as well. Otherwise the rep would be removed as soon as cleanup_reps is called.

Reply:
Is that just checking if a channel is inactive?
The rep crawler should not decide if a channel is inactive, the channel should have a method: ```
bool inactive()
```
and it should decided by itself if it active or not.

Reply:
Yes, thought so too. It is weird. A few lines up at line 295 there is a check whether a channel is closed by checking the endpoint for `any`, which means the channel was closed. I don't understand why there are these additional channel comparisons. Do you know why? The check in line 295 should be enough, shouldn't it?

Reply:
Found some more time to look at it... I think I understand what's going on now. The problem is, that the class `tcp_channels` removes inactive channels in its `purge()` method, but it does *not* change the state of the channel. So that's why we cannot ask the channel if it is closed. I believe it would be a good idea to add the methods `close()` and `is_closed()` to `channel`. `tcp_channels` should close the purged channels and `rep_crawler` should only check whether a channel is closed. That would simplify this code. I think these changes would belong to a new pull request, because the current pull request is focused on fixing the unit test. Thoughts?

Reply:
Yes, I think the changes you describe make sense and they should be a separate PR.
do you plan to work on them or would you like one of us to do it?

Reply:
If you guys have time for this then I could focus on the Rust stuff. If you are busy with other things then I would do it.

Reply:
No worries, we can take care of it.